### ISSUE TITLE
Applying profile with limits.cpu: "XX" set does not always update available CPU threads as seen inside the container

### ISSUE THREAD

**Issue by Kramerican** [lxc/lxc#3427]
Focal
LXD v4.0.1

**Symptoms:**

Creating two profiles with limits.cpu - say one called "bigprofile" with 

```
config:
  limits.cpu: "30"
```

And another "smallprofile" with

```
config:
  limits.cpu: "20"
```

And then you apply these to a running container with `lxc profile apply` you will - more often than not - find that if you then enter the container, the available cpu count as seen in tools such as `htop` has not taken effect. It may take a few attempts at applying the profile before seeing this failure happen.

**Other observations:**

A profile will always correctly update the available thread count for CPU as seen inside the container if you edit the profile and change the limits.cpu number with `lxc profile edit` (I have yet to see this fail anyway).

Restarting the container always brings up the correct thread count as defined in the profile.

If the `lxc profile apply` failed to update limits.cpu it will stay like that indefinitely untill you do a restart or otherwise force it to update (e.g. with a profile edit)

Access to systems where this is happening can be granted on request :)

**Comment by Kramerican** [comment]
`return "shit is busted"` rofl

Ok, dumb this down for me a little bit @stgraber - in my real world situation, as-is right now, what I should probably do is something like:

- Apply the profile
- Run `lxc exec containername nproc` and look at the output and see if it's now what I expect, if not do a container restart

Is that a reasonable "dumb but works" way of doing it untill all your awesome new cpuset sauce gets added to LXC?

**Comment by stgraber** [comment]
Yeah, that's your only way out for containers with needed sub-cgroups right now.
To minimize the number of those, you probably want to check for containers which have directories under /sys/fs/cgroup/cpuset and try to figure out where those come from.

If you only have files in /sys/fs/cgroup/cpuset, then cpu limit changes will work.
